tests/ashley/machina_extensions/forum_conversation/test_forum_conversation.py

```diff
@@ -0,0 +1,198 @@
+            {
+                "name": "Alfred",
+                "link": f"/forum/member/profile/{user2.id}/",
+            }
+        ]
+        # user3 loads the form
+        form = PostForm(data=form_data, user=user3, forum=self.forum, topic=topic)
+        # user3 sees user1 and user2 in the list of users to mention orderedy by username,
+        # user2 should be listed first as Alfred is before Benoit
+        assert form.fields["content"].widget.attrs["mention_users"] == [
+            {"name": "Alfred", "link": f"/forum/member/profile/{user2.id}/"},
+            {"name": "Benoit", "link": f"/forum/member/profile/{user1.id}/"},
+        ]
+        # user1 add a new post
+        PostFactory(subject="Réponse", topic=topic, poster=user1)
+        # Nothing should have changed as there is no new user that posted
+        form = PostForm(data=form_data, user=user3, forum=self.forum, topic=topic)
+        assert form.fields["content"].widget.attrs["mention_users"] == [
+            {"name": "Alfred", "link": f"/forum/member/profile/{user2.id}/"},
+            {"name": "Benoit", "link": f"/forum/member/profile/{user1.id}/"},
+        ]
+
+        # Set up new topic and initial post with user4
+        topic2 = TopicFactory(forum=self.forum, poster=user4)
+        post_topic2_user4 = PostFactory(subject="Réponse", topic=topic2, poster=user4)
+
+        # user3 loads the form for topic2
+        form = PostForm(data=form_data, user=user3, forum=self.forum, topic=topic2)
+        assert form.fields["content"].widget.attrs["mention_users"] == [
+            {
+                "name": "user4",
+                "link": f"/forum/member/profile/{user4.id}/",
+            }
+        ]
+        # user3 loads topic, nothing should have changed as user4 only posted in another topic
+        form = PostForm(data=form_data, user=user3, forum=self.forum, topic=topic)
+        assert form.fields["content"].widget.attrs["mention_users"] == [
+            {"name": "Alfred", "link": f"/forum/member/profile/{user2.id}/"},
+            {"name": "Benoit", "link": f"/forum/member/profile/{user1.id}/"},
+        ]
+        # Post of user4 gets unapproved
+        post_topic2_user4.approved = False
+        post_topic2_user4.save()
+        # Load topic2
+        form = PostForm(data=form_data, user=user3, forum=self.forum, topic=topic2)
+        # List of active users should be empty for topic2
+        assert form.fields["content"].widget.attrs["mention_users"] == []
+
+        # user2 become inactive
+        user2.is_active = False
+        user2.save()
+        # user3 loads the form for topic
+        form = PostForm(data=form_data, user=user3, forum=self.forum, topic=topic)
+        # We should only see user1 in the list of active users for this topic
+        assert form.fields["content"].widget.attrs["mention_users"] == [
+            {"name": "Benoit", "link": f"/forum/member/profile/{user1.id}/"},
+        ]
```

Comment:
This is really well written and robust and I'm fine merging it but maybe the reallybest practice would be to split it in x tests each proving one of the points. Each test would be simpler to understand and easier to maintain.

Reply:
ok Thank you Sam for the review and feedback I will do so